### ISSUE TITLE
Can'r compile with (Portable) Msys2

### ISSUE THREAD

**Issue by Samuel-DEVULDER** [bebbo/newlib-cygwin#5]
On a windows machine, amiga-gcc fails to compile newlib-cygwin using (Portable)MSys2 (https://portableapps.com/node/56214)

What I got is:
```
Utilisateur@W520 MSYS /tmp/amiga-gcc
$ make newlib
make newlib...failed
make[1] : on entre dans le répertoire « /tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/newlib »
make "AR_FLAGS=rc" "CC_FOR_BUILD=gcc" "CFLAGS=-O2 -fomit-frame-pointer" "CCASFLAGS=-O2 -fomit-frame-pointer" "CFLAGS_FOR_BUILD=" "CFLAGS_FOR_TARGET=" "INSTALL=/usr/bin/install -c" "LDFLAGS=" "LIBCFLAGS=" "LIBCFLAGS_FOR_TARGET=" "MAKE=make" "MAKEINFO=/bin/sh /tmp/amiga-gcc/projects/newlib-cygwin/missing --run makeinfo " "PICFLAG=" "PICFLAG_FOR_TARGET=" "SHELL=/bin/sh" "EXPECT=`if test -f ./../expect/expect; then echo ./../expect/expect; else echo expect; fi`" "RUNTEST=`if test -f /tmp/amiga-gcc/projects/newlib-cygwin/newlib/../dejagnu/runtest; then echo /tmp/amiga-gcc/projects/newlib-cygwin/newlib/../dejagnu/runtest; else echo runtest; fi`" "RUNTESTFLAGS=" "exec_prefix=/opt/amiga" "infodir=/opt/amiga/share/info" "libdir=/opt/amiga/lib" "prefix=/opt/amiga" "tooldir=/opt/amiga/m68k-amigaos" "top_toollibdir=/opt/amiga/m68k-amigaos/lib" "AR=m68k-amigaos-ar" "AS=m68k-amigaos-as" "CC=m68k-amigaos-gcc -I/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/newlib/targ-include -I/tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/include" "LD=" "LIBCFLAGS=" "NM=" "PICFLAG=" "RANLIB=m68k-amigaos-ranlib" "DESTDIR=" all-recursive
make[2] : on entre dans le répertoire « /tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/newlib »
make "AR_FLAGS=rc" "CC_FOR_BUILD=gcc" "CFLAGS=-O2 -fomit-frame-pointer" "CCASFLAGS=-O2 -fomit-frame-pointer" "CFLAGS_FOR_BUILD=" "CFLAGS_FOR_TARGET=" "INSTALL=/usr/bin/install -c" "LDFLAGS=" "LIBCFLAGS=" "LIBCFLAGS_FOR_TARGET=" "MAKE=make" "MAKEINFO=/bin/sh /tmp/amiga-gcc/projects/newlib-cygwin/missing --run makeinfo  " "PICFLAG=" "PICFLAG_FOR_TARGET=" "SHELL=/bin/sh" "EXPECT=expect" "RUNTEST=runtest" "RUNTESTFLAGS=" "exec_prefix=/opt/amiga" "infodir=/opt/amiga/share/info" "libdir=/opt/amiga/lib" "prefix=/opt/amiga" "tooldir=/opt/amiga/m68k-amigaos" "top_toollibdir=/opt/amiga/m68k-amigaos/lib" "AR=m68k-amigaos-ar" "AS=m68k-amigaos-as" "CC=m68k-amigaos-gcc -I/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/newlib/targ-include -I/tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/include" "LD=" "LIBCFLAGS=" "NM=" "PICFLAG=" "RANLIB=m68k-amigaos-ranlib" "DESTDIR=" DO=all multi-do # make
make[3] : on entre dans le répertoire « /tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/newlib »
if [ -z "libb libm020 libb/libm020 libb32/libm020" ]; then \
  true; \
else \
  rootpre=`${PWDCMD-pwd}`/; export rootpre; \
  srcrootpre=`cd /tmp/amiga-gcc/projects/newlib-cygwin/newlib; ${PWDCMD-pwd}`/; export srcrootpre; \
  lib=`echo "${rootpre}" | sed -e 's,^.*/\([^/][^/]*\)/$,\1,'`; \
  compiler="m68k-amigaos-gcc -I/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/newlib/targ-include -I/tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/include"; \
  for i in `${compiler} --print-multi-lib 2>/dev/null`; do \
    dir=`echo $i | sed -e 's/;.*$//'`; \
    if [ "${dir}" = "." ]; then \
      true; \
    else \
      if [ -d ../${dir}/${lib} ]; then \
        flags=`echo $i | sed -e 's/^[^;]*;//' -e 's/@/ -/g'`; \
        if (cd ../${dir}/${lib}; make "AR_FLAGS=rc" "CC_FOR_BUILD=gcc" "CFLAGS=-O2 -fomit-frame-pointer" "CCASFLAGS=-O2 -fomit-frame-pointer" "CFLAGS_FOR_BUILD=" "CFLAGS_FOR_TARGET=" "INSTALL=/usr/bin/install -c" "LDFLAGS=" "LIBCFLAGS=" "LIBCFLAGS_FOR_TARGET=" "MAKE=make" "MAKEINFO=/bin/sh /tmp/amiga-gcc/projects/newlib-cygwin/missing --run makeinfo   " "PICFLAG=" "PICFLAG_FOR_TARGET=" "SHELL=/bin/sh" "EXPECT=expect" "RUNTEST=runtest" "RUNTESTFLAGS=" "exec_prefix=/opt/amiga" "infodir=/opt/amiga/share/info" "libdir=/opt/amiga/lib" "prefix=/opt/amiga" "tooldir=/opt/amiga/m68k-amigaos" "top_toollibdir=/opt/amiga/m68k-amigaos/lib" "AR=m68k-amigaos-ar" "AS=m68k-amigaos-as" "CC=m68k-amigaos-gcc -I/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/newlib/targ-include -I/tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/include" "LD=" "LIBCFLAGS=" "NM=" "PICFLAG=" "RANLIB=m68k-amigaos-ranlib" "DESTDIR=" \
                        CFLAGS="-O2 -fomit-frame-pointer ${flags}" \
                        CCASFLAGS="-O2 -fomit-frame-pointer ${flags}" \
                        FCFLAGS=" ${flags}" \
                        FFLAGS=" ${flags}" \
                        ADAFLAGS=" ${flags}" \
                        prefix="/opt/amiga" \
                        exec_prefix="/opt/amiga" \
                        GCJFLAGS=" ${flags}" \
                        GOCFLAGS=" ${flags}" \
                        CXXFLAGS=" ${flags}" \
                        LIBCFLAGS=" ${flags}" \
                        LIBCXXFLAGS=" ${flags}" \
                        LDFLAGS=" ${flags}" \
                        MULTIFLAGS="${flags}" \
                        DESTDIR="" \
                        INSTALL="/usr/bin/install -c" \
                        INSTALL_DATA="/usr/bin/install -c -m 644" \
                        INSTALL_PROGRAM="/usr/bin/install -c" \
                        INSTALL_SCRIPT="/usr/bin/install -c" \
                        all); then \
          true; \
        else \
          exit 1; \
        fi; \
      else true; \
      fi; \
    fi; \
  done; \
fi
make[4] : on entre dans le répertoire « /tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/libb/newlib »
make "AR_FLAGS=rc" "CC_FOR_BUILD=gcc" "CFLAGS=-O2 -fomit-frame-pointer  -fbaserel" "CCASFLAGS=-O2 -fomit-frame-pointer  -fbaserel" "CFLAGS_FOR_BUILD=" "CFLAGS_FOR_TARGET=" "INSTALL=/usr/bin/install -c" "LDFLAGS=-fbaserel" "LIBCFLAGS=-fbaserel" "LIBCFLAGS_FOR_TARGET=" "MAKE=make" "MAKEINFO=/bin/sh /tmp/amiga-gcc/projects/newlib-cygwin/missing --run makeinfo    " "PICFLAG=" "PICFLAG_FOR_TARGET=" "SHELL=/bin/sh" "EXPECT=expect" "RUNTEST=runtest" "RUNTESTFLAGS=" "exec_prefix=/opt/amiga" "infodir=/opt/amiga/share/info" "libdir=/opt/amiga/lib" "prefix=/opt/amiga" "tooldir=/opt/amiga/m68k-amigaos" "top_toollibdir=/opt/amiga/m68k-amigaos/lib/libb" "AR=m68k-amigaos-ar" "AS=m68k-amigaos-as" "CC=m68k-amigaos-gcc  -fbaserel -I/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/libb/newlib/targ-include -I/tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/include" "LD=" "LIBCFLAGS=-fbaserel" "NM=" "PICFLAG=" "RANLIB=m68k-amigaos-ranlib" "DESTDIR=" all-recursive
make[5] : on entre dans le répertoire « /tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/libb/newlib »
true "AR_FLAGS=rc" "CC_FOR_BUILD=gcc" "CFLAGS=-O2 -fomit-frame-pointer  -fbaserel" "CCASFLAGS=-O2 -fomit-frame-pointer  -fbaserel" "CFLAGS_FOR_BUILD=" "CFLAGS_FOR_TARGET=" "INSTALL=/usr/bin/install -c" "LDFLAGS=-fbaserel" "LIBCFLAGS=-fbaserel" "LIBCFLAGS_FOR_TARGET=" "MAKE=make" "MAKEINFO=/bin/sh /tmp/amiga-gcc/projects/newlib-cygwin/missing --run makeinfo     " "PICFLAG=" "PICFLAG_FOR_TARGET=" "SHELL=/bin/sh" "EXPECT=expect" "RUNTEST=runtest" "RUNTESTFLAGS=" "exec_prefix=/opt/amiga" "infodir=/opt/amiga/share/info" "libdir=/opt/amiga/lib" "prefix=/opt/amiga" "tooldir=/opt/amiga/m68k-amigaos" "top_toollibdir=/opt/amiga/m68k-amigaos/lib/libb" "AR=m68k-amigaos-ar" "AS=m68k-amigaos-as" "CC=m68k-amigaos-gcc  -fbaserel -I/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/libb/newlib/targ-include -I/tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/include" "LD=" "LIBCFLAGS=-fbaserel" "NM=" "PICFLAG=" "RANLIB=m68k-amigaos-ranlib" "DESTDIR=" DO=all multi-do # make
Making all in libc
make[6] : on entre dans le répertoire « /tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/libb/newlib/libc »
Making all in argz
make[7] : on entre dans le répertoire « /tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/libb/newlib/libc/argz »
make[7]: rien à faire pour « all ».
make[7] : on quitte le répertoire « /tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/libb/newlib/libc/argz »
Making all in stdlib
make[7] : on entre dans le répertoire « /tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/libb/newlib/libc/stdlib »
m68k-amigaos-gcc  -fbaserel -I/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/libb/newlib/targ-include -I/tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/include -DPACKAGE_NAME=\"newlib\" -DPACKAGE_TARNAME=\"newlib\" -DPACKAGE_VERSION=\"3.0.0\" -DPACKAGE_STRING=\"newlib\ 3.0.0\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -I. -I/tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/stdlib -DCOMPACT_CTYPE -DSIGNAL_PROVIDED -fno-builtin      -O2 -fomit-frame-pointer  -fbaserel -c -o lib_a-dtoa.o `test -f 'dtoa.c' || echo '/tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/stdlib/'`dtoa.c
In file included from /tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/stdlib/dtoa.c:33:0:
/tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/stdlib/mprec.h:35:33: fatal error: ../locale/setlocale.h: No such file or directory
 #include "../locale/setlocale.h"
                                 ^
compilation terminated.
make[7]: *** [Makefile:730: lib_a-dtoa.o] Error 1
make newlib...failed
use less "log/make newlib.log" to view the full log and search for ***
make: *** [Makefile:891: build-MSYS_NT-10.0-WOW/newlib/newlib/libc.a] Error 2

```

From the look, it see that compilation is done in /tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/newlib/libb/newlib/libc/stdlib. Sofar so good, but there it fails to include "../locale/setlocale.h". This is logical since there is no such .h under the build folder. It appears that gcc was invoked with -I/tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/stdlib, but it wont search in there since the include is "../locale/setlocale.h" and not <../locale/setlocale.h>.

**Comment by Samuel-DEVULDER** [comment]
This is really odd. From the doc "..." include should search from the current source folder with should produce /tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/stdlib/../locale/setlocale.h which looks okay:
```
Utilisateur@W520 MSYS /tmp/amiga-gcc
$ ls -l /tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/stdlib/../locale/setlocale.h
-rw-r--r-- 1 Utilisateur Aucun 11942 17 sept. 19:20 /tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/stdlib/../locale/setlocale.h

```

hmm.. but gcc-amigaos doesn't see it!
```
Utilisateur@W520 MSYS /tmp/amiga-gcc
$ m68k-amigaos-gcc /tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/stdlib/../locale/setlocale.h
cc1: fatal error: /tmp/amiga-gcc/projects/newlib-cygwin/newlib/libc/stdlib/../locale/setlocale.h: No such file or directory
compilation terminated.

```

Cross-developpment nightmare :(

**Comment by Samuel-DEVULDER** [comment]
I'll try to recompile outside of MSys internal-path (e.g. on some /e/ drive of some kind) to see if it fixes the issue.

**Comment by bebbo** [comment]
seems like the path handling is broken somehow!?

What happens if you use a dos-like path, starting with a drive letter?  `c:/msys2/tmp/...`

**Comment by Samuel-DEVULDER** [comment]
Ok, this seem to do the trick. I have moved amiga-gcc out from /tmp to /c/tmp (eg outside of MSYS "path") and compilation goes further apparently[*].

So conclusion is: under MSYS2, better compile outoutside of internal filesystem-tree (eg. not in /tmp nor in /home/something, but go into /c/tmp or some other places outside out of msys filesystem tree).
____
[*] it fails at libgcc, but this is a different matter
```
make[4] : on entre dans le répertoire « /c/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/gcc/m68k-amigaos/libstdc++-v3/include »
mkdir -p ./m68k-amigaos/bits/stdc++.h.gch
/c/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/gcc/./gcc/xgcc -shared-libgcc -B/c/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/gcc/./gcc -nostdinc++ -L/c/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/gcc/m68k-amigaos/libstdc++-v3/src -L/c/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/gcc/m68k-amigaos/libstdc++-v3/src/.libs -L/c/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/gcc/m68k-amigaos/libstdc++-v3/libsupc++/.libs -B/opt/amiga/m68k-amigaos/bin/ -B/opt/amiga/m68k-amigaos/lib/ -isystem /opt/amiga/m68k-amigaos/include -isystem /opt/amiga/m68k-amigaos/ndk-include -isystem /opt/amiga/m68k-amigaos/sys-include    -x c++-header -nostdinc++ -O2 -fomit-frame-pointer  -I/c/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/gcc/m68k-amigaos/libstdc++-v3/include/m68k-amigaos -I/c/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/gcc/m68k-amigaos/libstdc++-v3/include -I/c/tmp/amiga-gcc/projects/gcc/libstdc++-v3/libsupc++  -O2 -g -std=gnu++0x /c/tmp/amiga-gcc/projects/gcc/libstdc++-v3/include/precompiled/stdc++.h \
-o m68k-amigaos/bits/stdc++.h.gch/O2ggnu++0x.gch
cc1plus: warning: the "stabs" debug format cannot be used with pre-compiled headers [-Wdeprecated]
In file included from /c/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/gcc/m68k-amigaos/libstdc++-v3/include/unordered_set:48:0,
                 from /c/tmp/amiga-gcc/projects/gcc/libstdc++-v3/include/precompiled/stdc++.h:118:
/c/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/gcc/m68k-amigaos/libstdc++-v3/include/bits/unordered_set.h:1434:1: internal compiler error: Segmentation fault
 } // namespace std
 ^

/c/tmp/amiga-gcc/build-MSYS_NT-10.0-WOW/gcc/m68k-amigaos/libstdc++-v3/include/bits/unordered_set.h:1434:1: internal compiler error: Bus error
mmap: No such device
Please submit a full bug report,
with preprocessed source if appropriate.
See <http://gcc.gnu.org/bugs.html> for instructions.
make[4]: *** [Makefile:1719: m68k-amigaos/bits/stdc++.h.gch/O2ggnu++0x.gch] Error 1
make libgcc...failed
use less "log/make libgcc.log" to view the full log and search for ***
make: *** [Makefile:785: build-MSYS_NT-10.0-WOW/gcc/_libgcc_done] Error 2

```

**Comment by bebbo** [comment]
closing since https://franke.ms/download/setup-amiga-gcc.exe is built day by day
